packages/contracts/contracts/voting/admin/Admin.sol

```diff
@@ -6,12 +6,13 @@ import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
 
 import {PluginCloneable} from "../../core/plugin/PluginCloneable.sol";
 import {Proposal} from "../../core/plugin/Proposal.sol";
+import {IMembership} from "../../core/plugin/IMembership.sol";
 import {IDAO} from "../../core/IDAO.sol";
 
 /// @title Admin
 /// @author Aragon Association - 2022.
-/// @notice The admin address governance plugin giving execution permission on the DAO to a single address.
-contract Admin is PluginCloneable, Proposal {
+/// @notice The admin governance plugin giving execution permission on the DAO to a single address.
+contract Admin is IMembership, PluginCloneable, Proposal {
```

Comment:
Sorry I know this doesn't belong in the scope of this pull but shouldn't we allow the admin address to be changeable?

Reply:
The admin power is granted via the `EXECUTE_PROPOSAL_PERMISSION_ID` during the setup. 
The reasoning was to not store the address in the contract (I would have preferred storing it, but in the end we preferred to save gas here).

Our design is: People should not manually grant / revoke permissions. If they do, it is their choice, but we cannot guarantee that installation / uninstallations work.
Another argument was that the `Admin` plugin is cheap and instead of changing the admin permission, people should uninstall the plugin and install a new `Admin` plugin.

So the answer is no, we planned it to not change. Ping at @Rekard0 how can give more details.

If we had wanted to change the admin address, we should have used a setter and explicitly store it.

Reply:
The reason for not storing it was not for saving gas, it was for allowing the admin address to be changeable or to have multiple admins if needed, and also for the moment of uninstalling we could not be fully sure if the stored admin is still the admin and have proper permission, or past admins has their permissions properly revoked.

In the design phase we explored the idea of using ownable by OZ as it is also transferable, but ultimately we decided against it, because using ownable was against our permission manager...etc

So currently the admin plugin is designed in a way it can have multiple admin, not even one, and those admins can be removed, added ...etc, while the DAO only give a set of permission (to operate on the DAO itself) to the admin-plugin and not the admins
And the admins have permission to operate on the plugin...etc
so this way once the plugin is uninstalled ,, it can not operate on the DAO, and it is not important which admin have permission on the plugin.

i hope that helped

Reply:
Oh, I wasn't aware. 
But this change of permissions collides with "never grant permissions manually".

So how would you suggest that we emit the `MembersAdded` and `MembersRemoved` event?

Reply:
After talking with @Rekard0, I came up with a nice fix for the problem: 7aed86e06a63b8dc19e74d29640f4ac580b73093

We inherit from the `IMembershipContract` interface instead and `Admin` implements
```solidity
/// @inheritdoc IMembershipContract
function isMember(address _account) external view returns (bool) {
    return
        dao.hasPermission({
            _where: address(this),
            _who: _account,
            _permissionId: EXECUTE_PROPOSAL_PERMISSION_ID,
            _data: bytes("")
        });
}
```

Reply:
How do we create members list then in the json ? 

1. if you grant manually, you can't catch the event ! so how would the flow work. here ? how would the json end up having 2 members in there ? how would `isMember` help here ?

Reply:
If the membership is based on a defining contract, such as a token, we just put the token or whatever defining contract in the member list. People that want to know who precisely is a member ask the `isMember` function.

Reply:
Agreed to be closed after discussion with @brickpop